Fix #4105: Fixed Enter Pin Text View in MDPI small

deepanshu731 opened the pull request:
Fixed #4105 
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
| Before | After |
|----------|---------|
|![Screenshot_20220214_225209](https://user-images.githubusercontent.com/78534813/154006214-a23fc8f4-6d63-4572-9c03-115f6a51aceb.png)|![Screenshot_20220214_221242](https://user-images.githubusercontent.com/78534813/154006285-2e3dfeb4-9b96-40d2-b3b5-215d2d889495.png)|
|![Screenshot_20220214_225230](https://user-images.githubusercontent.com/78534813/154006496-e4c058ac-9260-42c9-8119-48062d2c90a3.png)|![Screenshot_20220214_221229](https://user-images.githubusercontent.com/78534813/154006426-0a0e10d2-efe2-44b6-bebb-9eda64d7459b.png)|
|![Screenshot_20220214_225034](https://user-images.githubusercontent.com/78534813/154006754-2e338315-d4e2-4a58-b58e-e92b7d7a1ebe.png)|![Screenshot_20220214_221355](https://user-images.githubusercontent.com/78534813/154006835-d34455cb-1509-408e-961c-6453220674c9.png)|
|![Screenshot_20220214_225021](https://user-images.githubusercontent.com/78534813/154006888-f7d46f1b-6ee8-4625-902e-8fbc64041121.png)|![Screenshot_20220214_221345](https://user-images.githubusercontent.com/78534813/154006943-ce001c0f-2529-4971-85de-71d20e5b2c4f.png)|
|![Screenshot_20220214_225426](https://user-images.githubusercontent.com/78534813/154007259-aa1f0a67-c4b2-437c-83d4-b2addea918ed.png)|![Screenshot_20220214_221044](https://user-images.githubusercontent.com/78534813/154007195-fc73f443-f904-4fbc-976f-fe9d3efa1187.png)|
|![Screenshot_20220214_225416](https://user-images.githubusercontent.com/78534813/154007309-ace6dcfb-fecf-4922-95a3-390359d61bcf.png)|![Screenshot_20220214_221108](https://user-images.githubusercontent.com/78534813/154007341-ac785cd5-f0f1-4f12-bcb9-2914235556bf.png)|

deepanshu731 committed:
fixed bug

rt4914 commented on app/src/main/res/layout/pin_password_activity.xml:
```diff
@@ -56,146 +56,153 @@
     </FrameLayout>
   </androidx.constraintlayout.widget.ConstraintLayout>
-</layout>
+</layout>
```
EOF is missing, add one blank line at the end.

deepanshu731 replied:
done

rt4914 reviewed (changes requested):
Please add before/after screenshot in all qualifiers like 
mobile-portrait
mobile-landscape
tablet-portrait
tablet-landscape

So that we can be sure that there is no change in UI for other qualifiers.

deepanshu731 committed:
Update pin_password_activity.xml

rt4914 commented:
@deepanshu731 Assign this PR to me once its ready for review again.

deepanshu731 committed:
some-changes

deepanshu731 committed:
Merge branch 'remove-pin-text-view-bug' of https://github.com/deepanshu731/oppia-android into remove-pin-text-view-bug

deepanshu731 commented:
@rt4914 i have added enter pin edit text view and show pin image in a horizontal linear layout and added weight to make the UI and given constraints accordingly .
There are few changes in UI in landscape of mdpi (3.2 inch size) and  pixel 3a (5.5 inches size) , i think afterward UI is better as its more linearly aligned , so what are your thoughts on this ?

rt4914 commented:
@deepanshu731 Unassigning myself for now. As discussed you can coordinate with @rishidyno , @vrajdesai78 and @adizcode about how to work on such UIs which can be creating in one single xml file with the usage of `-1px` and `-2px` values.
